test/hotspot/jtreg/runtime/Monitor/MonitorUsedDeflationThresholdTest.java

```diff
@@ -0,0 +1,138 @@
+
+            System.out.println("Delaying for " + DELAY_SECS + " secs.");
+            try {
+                Thread.sleep(DELAY_SECS * 1000);
+            } catch (InterruptedException ie) {
+                // ignore InterruptedException
+            }
+            System.out.println("Done delaying for " + DELAY_SECS + " secs.");
+            return;
+        }
+
+        Object obj = new Object();
+
+        synchronized(obj) {
+            try {
+                obj.wait(1);  // force inflation
+            } catch (InterruptedException ie) {
+                // ignore InterruptedException
+            }
+            do_work(count + 1);
+        }
+    }
+
+    public static void usage() {
+        System.err.println("Usage: java " +
+                           "MonitorUsedDeflationThresholdTest inflate_count");
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            // Without args we invoke the test in a java sub-process:
+            ProcessBuilder pb = ProcessTools.createJavaProcessBuilder(
+                // Test doesn't need much Java heap:
+                "-Xmx100M",
+                // AvgMonitorsPerThreadEstimate == 1 means we'll start with
+                // an in_use_list_ceiling of <n-threads> plus a couple of
+                // of monitors for threads that call Object.wait().
+                "-XX:+UnlockDiagnosticVMOptions",
+                "-XX:AvgMonitorsPerThreadEstimate=1",
+                // Enable monitorinflation logging so we can see that
+                // MonitorUsedDeflationThreshold and
+                // NoAsyncDeflationProgressMaxoption are working.
+                "-Xlog:monitorinflation=info",
+                // Enable some safepoint logging for diagnostic purposes.
+                "-Xlog:safepoint+cleanup=info",
+                "-Xlog:safepoint+stats=debug",
+                // Run the test with inflate_count == 33 since that
+                // reproduced the bug with JDK13. Anything above the
+                // in_use_list_ceiling will do the trick.
+                "MonitorUsedDeflationThresholdTest", "33");
+
+            OutputAnalyzer output_detail = new OutputAnalyzer(pb.start());
+
+            // This mesg means:
+            // - AvgMonitorsPerThreadEstimate == 1 reduced in_use_list_ceiling
+            //   to a small number.
+            // - and we crossed MonitorUsedDeflationThreshold:
+            output_detail.shouldMatch("begin deflating: .*");
+            System.out.println("Found beginning of a deflation cycle.");
+
+            // This mesg means we hit NoAsyncDeflationProgressMax and
+            // had to adjust the in_use_list_ceiling:
+            String too_many = output_detail.firstMatch("Too many deflations without progress; .*", 0);
+            if (too_many == null) {
+                throw new RuntimeException("Did not find too_many string in output.\n");
+            }
```

Comment:
You should call output_detail.reportDiagnosticSummary() before throwing the exception. That emulates what OutputAnalyzer.shouldContain would do. (or you could just use shouldContain).

Reply:
I added a call to output_detail.reportDiagnosticSummary().
I didn't use OutputAnalyzer.shouldContain() intentionally
because of this output line:

`System.out.println("too_many='" + too_many + "'");`

which is there for diagnostic purposes.

Reply:
Manually tested the use of `output_detail.reportDiagnosticSummary()`
by temporarily setting `too_many = null;` before the check. It caused
nice diagnostic output to be printed. I like this technique!

Reply:
For the record, when the test passes, the output looks like this:

```
[2021-01-13T16:12:47.673259Z] Gathering output for process 15104
Found beginning of a deflation cycle.
too_many='Too many deflations without progress; bumping in_use_list_ceiling from 36 to 40'
PASSED.
----------System.err:(1/15)----------
STATUS:Passed.
----------rerun:(34/5039)*----------
```

so there's just a bit of diagnostic output from the test in the passing case.